silverback/persistence.py

```diff
@@ -0,0 +1,287 @@
+import pickle
+import sqlite3
+from abc import ABC, abstractmethod
+from datetime import datetime, timezone
+from typing import Optional
+
+from pydantic import BaseModel
+from taskiq import TaskiqResult
+from typing_extensions import Self  # Introduced 3.11
+
+from .types import IntOrNone, ISilverbackSettings, SilverbackIdent
+
+
+class SilverbackState(BaseModel):
+    instance: str
+    network: str
+    # Last block number seen by runner
+    last_block_seen: int
+    # Last block number processed by a worker
+    last_block_processed: int
+    updated: datetime
+
+
+class HandlerResult(BaseModel):
+    instance: str
+    network: str
+    handler_id: str
+    block_number: Optional[int]
+    log_index: Optional[int]
+    execution_time: float
+    # TODO: upcoming feature in taskiq
+    # labels: Dict[str]
+    # TODO: Use computed field with pydantic v2
+    return_value_blob: Optional[bytes]  # pickled data
+    created: datetime
+
+    @classmethod
+    def from_taskiq(
+        cls,
+        ident: SilverbackIdent,
+        handler_id: str,
+        block_number: IntOrNone,
+        log_index: IntOrNone,
+        result: TaskiqResult,
+    ) -> Self:
+        return cls(
+            instance=ident.identifier,
+            network=ident.network_choice,
+            handler_id=handler_id,
+            block_number=block_number,
+            log_index=log_index,
+            execution_time=result.execution_time,
+            # labels=result.labels,
+            return_value_blob=pickle.dumps(result.return_value),
```

Comment:
issue: pickling user data is dangerous.  Look into leveraging pydantic better here.